server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -164,36 +191,61 @@ private List<DateTimeFormatter> createRoundUpParser(
         return null;
     }
 
-    public static DateFormatter combined(String input, List<DateFormatter> formatters) {
+    public static DateFormatter combined(
+        String input,
+        List<DateFormatter> formatters,
+        DateFormatter printFormatter,
+        Boolean canCacheLastParsedFormatter
+    ) {
         assert formatters.size() > 0;
+        assert printFormatter != null;
 
-        List<DateTimeFormatter> parsers = new ArrayList<>(formatters.size());
+        List<DateTimeFormatter> parsers = new CopyOnWriteArrayList<>();
```

Comment:
```suggestion
        List<DateTimeFormatter> parsers = new ArrayList<>(formatters.size());
```
There is no race here, and `JavaDateFormatter` will wrap parsers into `CopyOnWriteArrayList` anyway (we don't need double wrapping).

Reply:
Are you sure we're double wrapping here? Because imo this function is calling `JavaDateFormatter` constructur just below this function where we're directly setting `this.parsers = parsers;` without wrapping?

Reply:
But this has to be fixed in `JavaDateFormatter` - it should not wrap in one place, but not in another

Reply:
Thanks, I got your point! Now, I've wrapped under both constructors